tests/suites/test_suite_ssl.function

```diff
@@ -224,34 +224,36 @@ int mbedtls_mock_socket_connect( mbedtls_mock_socket* peer1,
 {
     int ret = -1;
 
-    peer1->input = peer2->output =
```

Comment:
Commit message for "Fix memory allocation fail in TCP mock socket", English edit:
"Because two buffers were aliased too early in the code, it was possible that after an allocation failure, free() would be called twice for the same pointer."

Reply:
done